src/coreComponents/constitutive/fluid/PVTFunctions/PVTFunctionHelpers.hpp

```diff
@@ -0,0 +1,71 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 Total, S.A
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file PVTFunctionHelpers.hpp
+ */
+
+#ifndef GEOSX_CONSTITUTIVE_FLUID_PVTFUNCTIONS_PVTFUNCTIONHELPERS_HPP
+#define GEOSX_CONSTITUTIVE_FLUID_PVTFUNCTIONS_PVTFUNCTIONHELPERS_HPP
+
+namespace geosx
+{
+
+namespace constitutive
+{
+
+namespace PVTProps
+{
+
+struct PVTFunctionHelpers
+{
+
+  /**
+   * @brief Look for the expectedNames in the names provided by the user
+   * @param[in] inputNames phase or component names provided by the user
+   * @param[in] expectedNames expected names that can be accepted by GEOSX
+   * @param[inout] id the index of the phase or component that was found
+   * @return true is an expectedName was found in the inputNames, false otherwise
+   */
+  static bool
+  findName( array1d< string > const & inputNames,
+            array1d< string > const & expectedNames,
+            localIndex & id )
+  {
+    id = -1;
+    for( localIndex i = 0; i < inputNames.size(); ++i )
+    {
+      string const input = inputNames[i];
+      for( localIndex j = 0; j < expectedNames.size(); ++j )
+      {
+        if( input == expectedNames[j] )
+        {
+          id = i;
+          return true;
+        }
+      }
+    }
+    return false;
+  }
```

Comment:
I recognize a standard algorithm in this:
```suggestion
  template<typename InputRange, typename ExpectedRange>
  static bool
  findName( InputRange const & inputNames,
            ExpectedRange const & expectedNames,
            localIndex & id )
  {
    using std::begin;
    using std::end;
    auto it = std::find_first_of( begin(inputNames), end(inputNames), begin(expectedNames), end(expectedNames) );
    id = std::distance( begin(inputNames), it );
    return it != end(inputNames);
  }
```
The template parameters are just so that you can mix and match `array1d`s, `std::vector`s and c-arrays (the last two have convenient list initialization, see the other suggestion above).
Moreover, I'm not a fan of passing an output parameter by reference, would rather have it be a return value. So either returning a `std::pair<bool, localIndex>`; or even just returning a `localIndex`, assigning it to the member variable and then asserting `0 <= m_compIndex && m_compIndex < componentNames.size()`.
(feel free to accept what you like of these suggestions, or nothing)

Reply:
Here, I implemented what you suggested. I now return the `localIndex` that I assign to the member variable, and I use the check that you suggested as well.